proc/threads.go

```diff
@@ -323,3 +324,26 @@ func (thread *Thread) Scope() (*EvalScope, error) {
 	}
 	return locations[0].Scope(thread), nil
 }
+
+func (thread *Thread) SetCurrentBreakpoint(excluded *Breakpoint) error {
```

Comment:
I don't see the value of passing anything in here, this method should always set the current breakpoint based on the PC. In the only case above where something is passed in, it's after a step which should mean we're past the breakpoint anyway. If `FindBreakpoint` is returning a breakpoint when it should not, then that is a bug with that function and should be dealt with there. I think the issue you're trying to work around is the fact that `FindBreakpoint` will try and resolve a breakpoint at `pc - <bp instruction len>`. That is to compensate for hitting a software breakpoint in which the current PC will be `<bp addr> + <bp instruction len>`. That is likely causing the issue in this case, since we could be single stepping over single byte instruction, after which we would incorrectly resolve the old breakpoint.